### ISSUE TITLE
arXiv

### ISSUE THREAD

**Issue by davidar** [ipfs/archives#2]
http://arxiv.org

- [x] harvest [OAI metadata](http://gateway.ipfs.io/ipfs/QmdKjhpvnWYbqps2hFJJNSBSwDv4NKQyPnxkAFWmAuHiuD/)
- [x] process the raw metadata stream into a [nicer format](http://gateway.ipfs.io/ipfs/Qmew9KT7J1yrr3Gc2kfYmZH1czARrmjrGW5HkgnFE4pufM)
- [ ] make processed metadata available as a unixfs tree rather than a tarball (blocked on ipfs/go-ipfs#1324)
- [x] download PDFs from S3
- [ ] download source files from S3
- [ ] extract the creative commons licensed paper IDs and mirror the associated files
- [ ] automate the process of [syncing OAI](https://github.com/davidar/oai-sync) repos to IPFS, and keeping them up-to-date

**Comment by jbenet** [comment]
:clap: :clap: :clap: :clap: @davidar  on setting up [the IPFS mirror of the CC part of arxiv](https://gateway.ipfs.io/ipfs/QmYaFMPTgDkwp3vun6o3zfVv6gFwDWVSJWwZQHieXj1Ys6).

**Comment by rht** [comment]
Don't have enough bandwidth to do this, but wish to `cat` the source archives https://gateway.ipfs.io/ipfs/QmPFiQcfUPr9DYTJ7MPzY9ixxWkfK1oqQQEmQHS1zeD5PP and then process it to get the most used operators 1-gram / var names.

e.g. with 50 sources:
```
1. { 15425 590.54%
2. _ 9059 346.82%
3. ^ 7225 276.61%
4. - 4081 156.24%
5. + 3019 115.58%
6. = 2612 100.00%
7. \frac 1757 67.27%
8. \mu 1485 56.85%
9. \partial 1295 49.58%
10. \bar 1290 49.39%
```

If brevity of a notation comes with usage.

**Comment by davidar** [comment]
@rht what I'd really love to do is http://latexsearch.com

But more generally, running computations on large distributed datasets is an interesting problem. IPMR (map reduce) anyone? It would be cool if you could pay nodes (bitcoin?) to run some sandboxed code over their local blockstore and publish the results back to the ipfs network. @jbenet have you had any thoughts on this topic?

**Comment by davidar** [comment]
@rht Also, if you could give me the script to do this I'd be happy to run it on pollux

**Comment by rht** [comment]
(on latexsearch : it is too crude. it's like searching through latex "bytecodes" instead of the formulas it represent)

I will put the script here after I have modified it to walk through several files.
Currently it is https://ipfs.io/ipfs/QmQtLp5BxYVgo6miYz3EUAaoP32eqUcW6G8ywsiMxGB5vT

**Comment by davidar** [comment]
rht: I guess my interest is more academic, as in, has anybody written anything about this expression I've just derived?

Also, would be really cool to finalise this issue soon (and publicise it a little bit). @whyrusleeping what's the best way to proceed with adding lots of small files at the moment?

**Comment by rht** [comment]
But it would be better if there are specifiers e.g. if the expression is vectorized, approximate/exact, etc.

https://ipfs.io/ipfs/QmcH6av29y7fXBqifmEMU54vVuMYjpBFBdcQsDkWZK8Hbi should work with `python parselatex.py dirname`

**Comment by davidar** [comment]
@rht definitely, and that's exactly why an open source version of LaTeX search would be better ☺

Thanks, will run (and time) that when I get a chance

- [ ] run https://ipfs.io/ipfs/QmcH6av29y7fXBqifmEMU54vVuMYjpBFBdcQsDkWZK8Hbi

**Comment by davidar** [comment]
@rht https://ipfs.io/ipfs/QmeFkmYSPhn33hzEHxxqtvDNxHorW93iZFFMgZDQKrfeH4

> real    6m24.104s
> user    3m34.448s
> sys     1m14.448s

**Comment by jbenet** [comment]
We should get @rht access on pollux -- @rht what's your ssh pubkey? 



—

**Comment by rht** [comment]
That's too fast!

This would have taken ages if I were to do it by myself (or if anyone had done this before).

It appears that
- \sum / \int (map-reduce of +) is the most used operator (\partial doesn't count because it appears 3 or 4 times in a line). This could have been `matmul`, but unfortunately this is implicit in latex.
- \alpha is actually being used more than \mu.
- \rangle happens more often than \psi (by a small margin)

**Comment by rht** [comment]
@jbenet https://api.github.com/users/rht/keys, the one with id 12491781, as @lgierth suggested.

**Comment by whyrusleeping** [comment]
@davidar the IPMR thing sounds like it could be accomplished with ethereum and ipfs.

**Comment by whyrusleeping** [comment]
> @whyrusleeping what's the best way to proceed with adding lots of small files at the moment? Edit: use the tar adder? Are you able to do something like ipfs cat /ipfs/hash/archive.tar/foo/bar.txt?

The most reliable way is going to be to add the content offline. Even the tar add command will hit the file descriptor issue :/

**Comment by lgierth** [comment]
> We should get @rht access on pollux

Will take care of it this sprint: https://github.com/ipfs/infrastructure/issues/87

**Comment by jbenet** [comment]
> it would be cool if you could pay nodes (bitcoin?) to run some sandboxed code over their local blockstore and publish the results back to the ipfs network. @jbenet have you had any thoughts on this topic?

that's like phase 3 of the IPPlan. (we're in phase 1). we could run some experiments now with this. it shouldn't be too hard.

**Comment by davidar** [comment]
> The most reliable way is going to be to add the content offline. Even the tar add command will hit the file descriptor issue :/

@whyrusleeping I am, still too slow :(

Re ethereum, can you elaborate? I've played with it a little bit, and it seems incapable of doing any kind of interesting computation before heat death of the universe...

@jbenet ipplan3, cool! What's phase 2?

**Comment by davidar** [comment]
There's still a little bit of work that needs to be done, but I'm going to go ahead and call this the initial release of the arXiv archive:

https://ipfs.io/ipfs/QmfXH9XtP7xmoTH8WAp4HNSduqWMwLTH8B8TvbTkdgzNAa

:balloon: :fireworks:

**Comment by jbenet** [comment]
:star2: :stars: :balloon: :+1:

**Comment by NDuma** [comment]
Link to GitHub via [GitXiv.com](http://www.gitxiv.com)

**Comment by davidar** [comment]
@NDuma GitXiv looks really cool, thanks! cc @samim23 @mekarpeles

**Comment by rht** [comment]
@davidar Request for reproducible build script. The spec for the package metadata (which includes the build process) should be achievable at 25 GB-scale (~ one server-node) regardless of the complication for many server-nodes case.

**Comment by davidar** [comment]
@rht I'm not sure there is a script (yet), it was mostly a (semi-)manual process... :/

**Comment by rht** [comment]
I see. I have re-created the scholarpedia.org archive in a local branch, since this is ~1GB, fits within one node of a computer. I wasn't able to find the example `_Metadata.json` from @eminence since it looks like the content has been gc-ed.

**Comment by leerspace** [comment]
The hash for this on archives.ipfs.io (/ipfs/QmfXH9XtP7xmoTH8WAp4HNSduqWMwLTH8B8TvbTkdgzNAa) seems to have become unavailable. If anyone still has a copy I can pin it to my node which I plan to leave running indefinitely.

**Comment by whyrusleeping** [comment]
I may have a copy on my NAS box which is currently in a storage unit (Recently moved). If nobody comes forth with another copy in the next couple months, i'll venture out to get it and plug it into a fat pipe

**Comment by razum2um** [comment]
Seems like QmfXH9XtP7xmoTH8WAp4HNSduqWMwLTH8B8TvbTkdgzNAa is still unavailable, isn't it?

**Comment by Stebalien** [comment]
I've managed to find a piece of it. I'm going to see if I can find any more.
